### PR TITLE
Add MatrixType and TableType to Python

### PR DESCRIPTION
**Pull request by jigold**
I'd appreciate comments on where this file should go.

### CODE REVIEW

File: hail/python/hail/ir/types.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        self.col_type = col_type
+        self.row_key = row_key
+        self.row_type = row_type
+        self.entry_type = entry_type
+
+    def __str__(self):
+        col_key = ','.join([escape_id(k) for k in self.col_key])
+        row_key = ','.join([escape_id(k) for k in self.row_key])
+        return f"Matrix{{" \
+                f"global:{self.global_type._jtype.parsableString()}," \
```
Comment (tpoterba) [inline hail/python/hail/ir/types.py]: shouldn't these be the Python str methods?

Comment (tpoterba) [reply hail/python/hail/ir/types.py]: ah. Hmm.

Comment (jigold) [reply hail/python/hail/ir/types.py]: These will get fixed with #4841

**Comment by jigold** [comment]
Closing because this is not the right solution to fixing MatrixRead for BGEN import.

**Pull request closed without merging**